### PR TITLE
[WIP][NOTEST] Sprout - Try reducing memory usage

### PR DESCRIPTION
**Pull request by mfalesni**

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit e8bfadd37a0bed6c53f2d33fc3870f976cbd2df6:

Everything seems all right :smile: .
*CFME QE Bot*

File: sprout/celery_runner [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
 import os
 os.environ["RUNNING_UNDER_SPROUT"] = "true"  # To prevent importing unnecessary UI stuff
 
-from celery.__main__ import main
+import re                           # noqa
+import sys                          # noqa
+
+from celery.__main__ import main    # noqa
+
+# Preload some used modules in advance
```
Comment (psav) [inline sprout/celery_runner]: What advantage does preloading have?

File: utils/appliance.py [hunk 2]
```diff
@@ -38,7 +36,9 @@
 RUNNING_UNDER_SPROUT = os.environ.get("RUNNING_UNDER_SPROUT", "false") != "false"
 # Do not import the whole stuff around
 if not RUNNING_UNDER_SPROUT:
-    from cfme.configure.configuration import set_server_roles, get_server_roles
```
Comment (psav) [inline utils/appliance.py]: This is to streamline it when we are not running under sprout, but is there protection on the methods that call using those imports?

File: utils/providers.py [hunk 3]
```diff
@@ -19,21 +20,26 @@
...
-    OpenstackInfraProvider, RHEVMProvider, VMwareProvider, SCVMMProvider)
-from cfme.middleware.provider import HawkularProvider
 from fixtures.prov_filter import filtered
 from utils import conf, version
 from utils.mgmt_system import RHEVMSystem
 from utils.log import logger, perflog
 from utils.wait import wait_for
 
+
+RUNNING_UNDER_SPROUT = os.environ.get("RUNNING_UNDER_SPROUT", "false") != "false"
```
Comment (psav) [inline utils/providers.py]: Same as question above.

**Comment by mfalesni** [comment]
Some modules limit loading of some other modules to conserve memory if running under sprout because sprout does not use the UI functionality for example. But thanks to the development, I guess it has already crept so that is why I polished it in the PR.

Preload is intended to preload the modules before forking so they don't need to be loaded again in all the workers. But I guess that won't work properly since python's refcounting is not so good for CoW memory management ...

**Comment by psav** [comment]
So, remove the preload and we'll merge the other UI removals?

**Comment by mfalesni** [comment]
@psav Possibly, but I am not sure :)

**Pull request closed without merging**